habitat/core/vector_env.py

```diff
@@ -294,63 +317,57 @@ def _spawn_workers(
             ps.daemon = True
             ps.start()
             worker_conn.close()
-        return (
-            [p.recv for p in parent_connections],
-            [p.send for p in parent_connections],
-        )
+
+        read_fns = [ReadWrapper(p.recv) for p in parent_connections]
```

Comment:
As I understand, you made blocking working process wise. That means you can send another command to the fastest process while others are still calculating the first command.

Reply:
Yep, this just makes the tracking of whether or not a process is doing work or is ready for me done per process instead of all processes.  The `_is_waiting` is also used by the write wrapper and read wrapper to make sure the write/read is valid and won't cause otherwise hard to debug bugs.

Reply:
I see. That's really nice.